src/astro_databricks/plugins/plugin.py

```diff
@@ -165,19 +250,37 @@ def get_link(
         *,
         ti_key: TaskInstanceKey | None = None,
     ) -> str:
-        dag = get_airflow_app().dag_bag.get_dag(ti_key.dag_id)
+        ti = None
+        if not ti_key:
+            ti = get_task_instance(operator, dttm)
+            ti_key = ti.key
+        task_group = get_task_group(operator)
+
+        dag = _get_flask_app().dag_bag.get_dag(ti_key.dag_id)
         dag.get_task(ti_key.task_id)
         self.log.info("Getting link for task %s", ti_key.task_id)
         if ".launch" not in ti_key.task_id:
             self.log.debug(
                 "Finding the launch task for job run metadata %s", ti_key.task_id
             )
-            ti_key = _get_launch_task_key(ti_key, group_id=operator.task_group.group_id)
+            launch_task_id = task_group.get_child_by_label("launch").task_id
+            ti_key = _get_launch_task_key(ti_key, task_id=launch_task_id)
         # Should we catch the exception here if there is no return value?
-        metadata = XCom.get_value(
-            ti_key=ti_key,
-            key="return_value",
-        )
+        try:
+            metadata = XCom.get_value(
+                ti_key=ti_key,
+                key="return_value",
+            )
+        except AttributeError:
```

Comment:
It's worth to add a comment in which circumstances we expect this to happen (Airflow version)

Reply:
yes, added